src/GitHub.VisualStudio/Menus/MenuProvider.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections.Generic;
+using System.ComponentModel.Composition;
+
+namespace GitHub.VisualStudio.Menus
+{
+    [Export(typeof(IMenuProvider))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class MenuProvider : IMenuProvider
+    {
+        [ImportMany]
+        public IEnumerable<IMenuHandler> Menus { get; set; }
```

Comment:
I'm a little wary of making these settable properties. Could we change the interface to make these read only properties and use ctor injection here like so?

```csharp
[ImportingConstructor]
public MenuProvider([ImportMany]IEnumerable<IMenuHandler> menus, [ImportMany]IEnumerable<IDynamicMenuHandler> dynamicMenus)
{
    // ...
}
```

I believe that better represents our intentions here.

Also, I'm guessing we should expose these properties as `IReadOnlyCollection<T>` rather than `IEnumerable<T>` since these aren't lazy enumerables and we want to avoid any possibility of multiple enumerations.